src/onchain/utils.ts

```diff
@@ -95,3 +97,71 @@ export function safeJSONStringify(obj: unknown): string {
     (_key, value) => (typeof value === "bigint" ? value.toString() : value) // return everything else unchanged
   );
 }
+
+export type PlutusDatum =
+  | string
+  | number
+  | bigint
+  | Buffer
+  | Map<PlutusDatum, PlutusDatum>
+  | PlutusDatum[]
+  | ConstrData;
+
+export class ConstrData {
+  constructor(public constr: number, public fields: PlutusDatum[]) {}
+}
+
+const DatumDecoderOptions = {
+  encoding: "hex" as const,
+  tags: Object.fromEntries(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    _.range(121, 128).map((tag) => [tag, (val: any) => new ConstrData(tag - 121, val)])
+  ),
+};
+
+export function decodeDatumSync(datum: string): PlutusDatum {
+  return cbor.decodeFirstSync(datum, DatumDecoderOptions);
+}
+
+export type ObjectPlutusDatum =
+  | string
+  | number
+  | bigint
+  | Buffer
+  | ObjectPlutusDatum[]
+  | Dictionary<ObjectPlutusDatum>;
+
+export function parseDatumLossy(
```

Comment:
This is just a temporary simplification. Ideally use something like `zod` to validate the token metadata